### PR TITLE
[COLLABORATION] Added architecture for add/delete file events

### PR DESCRIPTION
**Pull request by hkirat**

### CODE REVIEW

**Comment by hkirat** [comment]
In reference to [#2353](https://github.com/mozilla/thimble.mozilla.org/issues/2353)

File: src/editor/Collaboration.js [hunk 1]
```diff
@@ -41,6 +43,24 @@ define(function (require, exports, module) {
 
             _webrtc.connection.on('message', _handleMessage);
         });
+
+        FileSystemEntry.on("change", function(event, entry, added, removed) {
+            if(added) {
```
Comment (gideonthomas) [inline src/editor/Collaboration.js]: import Filer.Path

```js
var root = StartupState.project("root");

if(added) {
    added.forEach(function(addedPath) {
        _webrtc.sendToAll("file-added", path.relative(addedPath, root));
    });
}

if(removed) {
    removed.forEach(function(removedPath) {
        _webrtc.sendToAll("file-removed", path.relative(removedPath, root));
    });
}

File: src/editor/Collaboration.js [hunk 2]
```diff
@@ -55,6 +75,16 @@ define(function (require, exports, module) {
             case "codemirror-change":
                 _handleCodemirrorChange(msg.payload);
                 break;
+            case "file-added":
+                var relPath = msg.payload;
+                var fullPath = StartupState.project("root") + relPath;
```
Comment (gideonthomas) [inline src/editor/Collaboration.js]: `path.join(root, msg.payload)`

File: src/editor/Collaboration.js [hunk 3]
```diff
@@ -55,6 +75,16 @@ define(function (require, exports, module) {
...
+                console.log("added file  " + fullPath);
+                break;
+            case "file-removed":
+                var relPath = msg.payload;
+                var fullPath = StartupState.project("root") + relPath;
```
Comment (gideonthomas) [inline src/editor/Collaboration.js]: same as above

**Comment by hkirat** [comment]
Up for review @gideonthomas

**Comment by hkirat** [comment]
Closing this against #858

**Pull request closed without merging**